### ISSUE TITLE
Insertion of too many rows throws exception with latest version

### ISSUE THREAD

**Issue by jaswanthbellam** [pgjdbc/r2dbc-postgresql#360]
This is a duplicate of https://github.com/pgjdbc/r2dbc-postgresql/issues/222 which was resolved earlier
Below is the code
```
final Statement insertOnConflict = connection.createStatement(insertOnConflictSQL);
Flux<Result> insertOnConflictFlux = Flux.from(insertOnConflict.execute());
connection.beginTransaction().then(
            insertOnConflictFlux
                    .map(result -> result.map((row, metadata) -> {
                        /**<code>**/
                    }))
                    .flatMap(Mono::from)
                    .collectList()
                    .delayUntil(val -> connection.commitTransaction())
                    .onErrorResume(t -> connection.rollbackTransaction().then(Mono.error(t)))
                    .doFinally(val -> pooledRef.release().subscribe())).then(Mono.empty());
});
```
Below is the exception. It work fine for less than 256 records. (was using version 0.8.4 the updated to latest 0.8.6)
```
{"@apptimestamp":"2020-12-03T19:38:14.342Z","level":"INFO","logger":"c.b.b.s.boot.logging.LoggingFilter","message":"Serving '/bifrost/app/refreshIMR'","environmentName":"UNSET","groupId":"GROUP","applicationId":"SERVICE","thread":"http-nio-8082-exec-1"}
2020-12-03 13:41:51,668 ERROR [reactor-tcp-nio-1] ChannelOperationsHandler: [id: 0x4d44bc82, L:/127.0.0.1:50642 - R:localhost/127.0.0.1:26000] Error was received while reading the incoming data. The connection will be closed.
reactor.core.Exceptions$ErrorCallbackNotImplemented: io.r2dbc.postgresql.client.ReactorNettyClient$RequestQueueException: Cannot exchange messages because the request queue limit is exceeded
Caused by: io.r2dbc.postgresql.client.ReactorNettyClient$RequestQueueException: Cannot exchange messages because the request queue limit is exceeded
	at io.r2dbc.postgresql.client.ReactorNettyClient$BackendMessageSubscriber.lambda$addConversation$2(ReactorNettyClient.java:768)
	at reactor.core.publisher.FluxCreate.subscribe(FluxCreate.java:94)
	at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:62)
	at io.r2dbc.postgresql.util.FluxDiscardOnCancel.subscribe(FluxDiscardOnCancel.java:49)
	at reactor.core.publisher.FluxDefer.subscribe(FluxDefer.java:54)
	at io.r2dbc.postgresql.util.FluxDiscardOnCancel.subscribe(FluxDiscardOnCancel.java:49)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4213)
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.drain(MonoIgnoreThen.java:172)
	at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:56)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4213)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:97)
	at reactor.core.publisher.MonoDelayUntil$DelayUntilCoordinator.onError(MonoDelayUntil.java:175)
	at reactor.core.publisher.MonoCollectList$MonoCollectListSubscriber.onError(MonoCollectList.java:106)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.checkTerminated(FluxFlatMap.java:834)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drainLoop(FluxFlatMap.java:600)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drain(FluxFlatMap.java:580)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onError(FluxFlatMap.java:443)
	at reactor.core.publisher.FluxMap$MapSubscriber.onError(FluxMap.java:126)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.checkTerminated(FluxFlatMap.java:834)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drainLoop(FluxFlatMap.java:600)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drain(FluxFlatMap.java:580)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.innerError(FluxFlatMap.java:855)
	at reactor.core.publisher.FluxFlatMap$FlatMapInner.onError(FluxFlatMap.java:1006)
	at reactor.core.publisher.FluxMap$MapSubscriber.onError(FluxMap.java:126)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onError(MonoFlatMapMany.java:197)
	at reactor.core.publisher.Operators$MonoSubscriber.onError(Operators.java:1829)
	at reactor.core.publisher.MonoCacheTime$CoordinatorSubscriber.signalCached(MonoCacheTime.java:323)
	at reactor.core.publisher.MonoCacheTime$CoordinatorSubscriber.onError(MonoCacheTime.java:346)
	at reactor.core.publisher.Operators$MonoSubscriber.onError(Operators.java:1829)
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreInner.onError(MonoIgnoreThen.java:235)
	at reactor.core.publisher.FluxHandle$HandleSubscriber.onError(FluxHandle.java:196)
	at reactor.core.publisher.FluxPeekFuseable$PeekConditionalSubscriber.onError(FluxPeekFuseable.java:894)
	at reactor.core.publisher.FluxCreate$BaseSink.error(FluxCreate.java:452)
	at reactor.core.publisher.FluxCreate$BufferAsyncSink.drain(FluxCreate.java:781)
	at reactor.core.publisher.FluxCreate$BufferAsyncSink.error(FluxCreate.java:726)
	at reactor.core.publisher.FluxCreate$SerializedSink.drainLoop(FluxCreate.java:229)
	at reactor.core.publisher.FluxCreate$SerializedSink.drain(FluxCreate.java:205)
	at reactor.core.publisher.FluxCreate$SerializedSink.error(FluxCreate.java:181)
	at io.r2dbc.postgresql.client.ReactorNettyClient$BackendMessageSubscriber.lambda$addConversation$2(ReactorNettyClient.java:768)
	at reactor.core.publisher.FluxCreate.subscribe(FluxCreate.java:94)
```

**Comment by mp911de** [comment]
The client is overloaded. The message `Cannot exchange messages because the request queue limit is exceeded` basically says that there are too many active client commands. There's nothing we can really do here as command subscriptions are controlled externally by your code calling the API code. We need to put subscriptions somewhere and the request queue is bounded.

**Comment by osi** [comment]
I'm also able to provoke this. It was surprising, as I would have thought that back pressure would have prevented me from submitting too many commands. Absent back pressure to communicate the status of the queue, how is application code expected to not hit this?

**Comment by cambierr** [comment]
any news here ? Main question being "how are we supposed to handle this then" ?

**Comment by mp911de** [comment]
I suggest the use of a connection pool. You can use pipelining over a single connection but at your own discretion and by ensuring request rate limiting. Other than that, we cannot do anything.

**Comment by cambierr** [comment]
The issue here is that in my case I'm in a transactional context - this means a single connection anyway.

Thus, I'm a bit unsure on how to manage that unless I'm missing something.. mauybe reduce `.flatMap` 's concurrency

**Comment by mp911de** [comment]
Please go for `concatMap` as that's processing items one by one. In JDBC, you would have one-by-one processing as well.

**Comment by cambierr** [comment]
forgot about that one ! thanks

**Comment by Muenze** [comment]
Hey, I have a similar problem:

We are using spring boot with r2dbc for postgre and kotlin lang.
I have an order event, a list of package events (in my case now 500) and address events, these need to be stored in 1 transaction.
My first approach was to use it with kotlin, map the package events wo be async {} and then awaitAll ... which was too much for the request queue

Then we switched from that approach to a Flux approach, now the 500 items were in that flux and I used saveAll ... again Exception on request queue

Then I found out about this issue page here, so I switched to flux.flatMap{ save(it) } .... exception on request queue
flux.concatMap { save(it) } ... exception on request queue

then I removed @Transactional => works like a charm

Is there a way I can set my queue size to be higher than 500, maybe 600 or 700? 
I saw in a different chat that someone posted: reactor.request....small = 10000 (don't know the exact property anymore), I used that for my integration test but still got the request limit exception.
